invite/app.ts

```diff
@@ -20,100 +20,86 @@ import {Octokit} from '@octokit/rest';
 
 import {InviteBot} from './src/invite_bot';
 
+type CommentWebhookPayload =
+  | Webhooks.WebhookPayloadIssueComment
+  | Webhooks.WebhookPayloadIssues
+  | Webhooks.WebhookPayloadPullRequest
+  | Webhooks.WebhookPayloadPullRequestReview
+  | Webhooks.WebhookPayloadPullRequestReviewComment;
+
 module.exports = (app: Application) => {
   if (process.env.NODE_ENV !== 'test') {
     require('dotenv').config();
   }
 
   const helpUserToTag = process.env.HELP_USER_TO_TAG || null;
 
-  app.on('issue_comment.created', async (
-    {github, event, payload, log}: Context<Webhooks.WebhookPayloadIssueComment>
-  ) => {
-    const inviteBot = new InviteBot(
+  function botFromContext(
+    {github, payload, log}: Context<CommentWebhookPayload>
+  ) {
+    return new InviteBot(
+      // This type-cast is required because Probot exports a separate GitHubAPI
+      // class, even though it's in Octokit instance.
       (github as unknown) as Octokit,
```

Comment:
Did you consider changing the original constructor to take a Context<CommentWebhookPayload> instead of an Octokit? :P

Reply:
I'd prefer not to tie the InviteBot constructor to Probot

Reply:
Fair 🤷‍♂